libevmasm/Assembly.h

```diff
@@ -47,9 +48,9 @@ using AssemblyPointer = std::shared_ptr<Assembly>;
 
 class Assembly
 {
-	using TagRefs = std::map<size_t, std::pair<size_t, size_t>>;
+	using TagRefs = std::map<size_t, std::pair<SubAssemblyID, size_t>>;
```

Comment:
We use `size_t` for tag ID but that should really be `uint64_t` as well. Would be nice to change that too at some point.

The tag+sub pair would also be better off as a proper struct with methods for converting to/from `u256`. We wouldn't then have to hard-code all those masks and 64s all over the place.

Reply:
Agree, it would also make it easier to reason about the whole program and data flow.